.gitignore

```diff
@@ -5,6 +5,7 @@
 *.iml
 /traefik
 /traefik.toml
+vendor/
```

Comment:
Point 1 is the tricky one. I've already had three attempts at writing this PR and in the end it seemed more satisfactory to complete the implementation instead of feeding small parts of it one by one. I'm open to persuasion though.